newm-server/src/main/kotlin/io/newm/server/aws/s3/model/Condition.kt

```diff
@@ -0,0 +1,147 @@
+package io.newm.server.aws.s3.model
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+@Serializable(with = Condition.Serializer::class)
+sealed class Condition {
+    abstract val key: String
+
+    object Serializer : KSerializer<Condition> {
```

Comment:
Do we really need a custom serializer in this case? I've done polymorphic patterns withe sealed classes before  without needing custom serialization, based on this:
https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/polymorphism.md

Reply:
I tried a couple of approaches and they all inserted the `type` discriminator key which breaks the functionality. I thought about trying some hacky stuff like stripping the `type` discriminator after the JSON was generated or adding a toJson field to the Condition object.  Happy to try other suggestions as well.